src/test/javascript/spec/integration/code-editor/code-editor-container.integration.spec.ts

```diff
@@ -164,37 +158,29 @@ describe('CodeEditorContainerIntegration', () => {
-                    subscribeForLatestResultOfParticipationSubject,
-                );
-                getFeedbackDetailsForResultStub = stub(resultService, 'getFeedbackDetailsForResult');
-                getBuildLogsStub = stub(buildLogService, 'getBuildLogs');
-                getFileStub = stub(codeEditorRepositoryFileService, 'getFile');
-                saveFilesStub = stub(codeEditorRepositoryFileService, 'updateFiles');
-                commitStub = stub(codeEditorRepositoryService, 'commit');
-                getStudentParticipationWithLatestResultStub = stub(programmingExerciseParticipationService, 'getStudentParticipationWithLatestResult');
-                getLatestPendingSubmissionStub = stub(submissionService, 'getLatestPendingSubmissionByParticipationId').returns(getLatestPendingSubmissionSubject);
+                checkIfRepositoryIsCleanStub = jest.spyOn(codeEditorRepositoryService, 'getStatus');
+                getRepositoryContentStub = jest.spyOn(codeEditorRepositoryFileService, 'getRepositoryContent');
+                subscribeForLatestResultOfParticipationStub = jest
+                    .spyOn(participationWebsocketService, 'subscribeForLatestResultOfParticipation')
+                    .mockReturnValue(subscribeForLatestResultOfParticipationSubject);
+                getFeedbackDetailsForResultStub = jest.spyOn(resultService, 'getFeedbackDetailsForResult');
+                getBuildLogsStub = jest.spyOn(buildLogService, 'getBuildLogs');
+                getFileStub = jest.spyOn(codeEditorRepositoryFileService, 'getFile');
+                saveFilesStub = jest.spyOn(codeEditorRepositoryFileService, 'updateFiles');
+                commitStub = jest.spyOn(codeEditorRepositoryService, 'commit');
+                getStudentParticipationWithLatestResultStub = jest.spyOn(programmingExerciseParticipationService, 'getStudentParticipationWithLatestResult');
+                getLatestPendingSubmissionStub = jest.spyOn(submissionService, 'getLatestPendingSubmissionByParticipationId').mockReturnValue(getLatestPendingSubmissionSubject);
             });
     });
 
     afterEach(() => {
-        checkIfRepositoryIsCleanStub.restore();
-        getRepositoryContentStub.restore();
-        subscribeForLatestResultOfParticipationStub.restore();
-        getFeedbackDetailsForResultStub.restore();
-        getBuildLogsStub.restore();
-        getFileStub.restore();
-        saveFilesStub.restore();
-        commitStub.restore();
-        getStudentParticipationWithLatestResultStub.restore();
+        jest.restoreAllMocks();
 
         subscribeForLatestResultOfParticipationSubject = new BehaviorSubject<Result | undefined>(undefined);
-        subscribeForLatestResultOfParticipationStub.returns(subscribeForLatestResultOfParticipationSubject);
+        subscribeForLatestResultOfParticipationStub.mockReturnValue(subscribeForLatestResultOfParticipationSubject);
```

Comment:
Setting these mocks in afterEach seems strange. Wouldn't it also work if they are set in the beforeEach?

Reply:
Yes it works in the beforeEach. I adapted it.